driver/src/price_estimation/dexag/mod.rs

```diff
@@ -54,21 +53,34 @@ where
 
 impl<Api> PriceSource for DexagClient<Api>
 where
-    Api: DexagApi,
+    Api: DexagApi + Sync,
 {
     fn get_prices(&self, tokens: &[Token]) -> Result<HashMap<TokenId, u128>> {
-        Ok(tokens
-            .iter()
-            .filter_map(|token| {
-                let stable_coin_price = if token.symbol() == self.stable_coin.symbol {
-                    1.0
-                } else {
-                    let dexag_token = self.tokens.get(token.symbol())?;
-                    self.api.get_price(&self.stable_coin, dexag_token).ok()?
-                };
-                Some((token.id, token.get_owl_price(stable_coin_price)))
-            })
-            .collect())
+        use rayon::prelude::*;
+        // Use rayon as a quick and dirty way to make requests in parallel. It
+        // would be more efficient if we used async or Futures instead but for
+        // now this is fine.
+        // async is not currently supported in traits so we cannot use it in the
+        // `PriceSource` trait.
+        let pool = rayon::ThreadPoolBuilder::new()
+            .num_threads(8)
+            .build()
+            .unwrap();
+        let result = pool.install(|| {
+            tokens
+                .par_iter()
+                .filter_map(|token| {
+                    let stable_coin_price = if token.symbol() == self.stable_coin.symbol {
```

Comment:
This variable stable_coin_price is only the stable coin price in one case, right? Maybe we can just call it 'price' or 'price_in_float_representation'

Reply:
We get some tokens as input for which we want the price in terms of owl. Every token is compared to `self.stable_coin` which acts as an approximation of owl. The reason there is an `if` is that it is possible that we are asked for the price token itself that we have picked as the stable coin. In this case it doesn't make sense to ask Dexag because it is the same coin. The price has to be 1.

Reply:
Maybe it is just me, but if I am hearing stable_coin_price, I think of the price of a stable coin. but what we actually have is a price of an arbitrary token vs a stable coin. So maybe price_vs_stable_coin makes more sense?

Reply:
Changed the variable name.